content/articles/2019-10-styler-1.2.0.Rmarkdown

```diff
@@ -0,0 +1,228 @@
+
+```{r}
+x <- 2
+x2 <- f(x)
+
+```
+
+unless you also use `strict = FALSE`.^[E.g. `styler::style_text(..., strict =
+FALSE)`, but note that this also has other effects on styling that you might not
+want.]
+
+# rlang's {{
+
+In [rlang 0.4.0](https://www.tidyverse.org/articles/2019/06/rlang-0-4-0/) a new
+interpolation operator was introduced: `{{` (read curly curly). Because `styler
+< 1.2.0` understood these as regular curly braces, you ended up with
+
+```{r}
+call({
+  {
+    x
+  }
+})
+```
+
+when styling `call({{x}})`, which is non-sense. Now styler yields `call({{ x
+}})`.
+
+# Addins
+
+The *Style active file* Addin now remembers the cursor position and the details
+of styling can be specified. For example, if you always only want to style
+spaces with the Addin, put this in your `.Rprofile`:^[E.g. via
+`usethis::edit_r_profile()`, making sure you use the right value for
+the `scope` argument]
+
+```{r}
+options(
+  styler.addins_style_transformer = "styler::tidyverse_style(scope = 'spaces')"
+)
+```
+
+You can also set the value of this option interactively with the *set style*
+Addin (not persistent over sessions). For details, see `help("styler_addins")`.
+
+The customization of the styling does not affect the command-line API
+(`styler::style_text()` and friends). We are not sure how users could best
+customize styling, but you can track progress on that in
+[r-lib/styler#319](https://github.com/r-lib/styler/issues/319).
+
+Remember that you can also set the environment variable `save_after_styling` to
+`TRUE` if you are tired of saving the file after styling it with the Addin.
+
+# Braces in function calls
+
+`tryCatch()` expressions often look like this:
+
+```{r}
+tryCatch(
+  {
+    exp(x)
+  },
+  error = function(x) x
+)
+```
+
+Prior to version 1.2.0, styler would return this odd formatting:
+
+```{r}
+tryCatch({
+  exp(x)
+},
+error = function(x) x
+)
+```
+
+Now, the line is broken before the opening curly brace in function calls, except
+if there is only one brace expression and it's the last in the function call.
+The typical use case is `testthat::test_that(...)`, i.e. the following code
+won't be modified:
+
+```{r}
+test_that("some condition holds", {
+  some_code()
+})
+```
+
+# Other changes
+
+Some other changes are:
+
+* styler depends on tibble >= 1.4.2 and runs 2x as fast as initially.
+
+* styler can style roxygen code examples in the source code of
+  packages.
+
+* styler can style `.Rnw` files.
+
+* The print method for the output of `style_text()` returns
+  syntax-highlighted code by default, controllable via the option
+  `styler.colored_print.vertical`.
+
+# Adaption of styler
+
+We'd like to highlight that styler integrates with various other tools you might
+be using:
+
+- As a git pre-commit hook. Two standard calls from the R console, and you are
+  all set. We are convinced that this is the preferred way of using styler to
+  ensure all your files are consistently formatted. Check out the [online
```

Comment:
To match our this this should be: Check out the [precommit](https://lorenzwalthert.github.io/precommit/) package for this and many other useful hooks.